<launch>

  <arg name="multimap_file" default="$(find utexas_gdc)/maps/simulation/multimap/2_only_multimap.yaml" />

  <!-- start the simulation environment -->
  <include file="$(find utexas_gdc)/launch/simulation_2.launch" />

  <!-- launch the multi map server for the simulated world -->
  <node name="multi_level_map_server" pkg="multi_level_map_server" type="multi_level_map_server">
    <param name="map_file" value="$(arg multimap_file)" />
  </node>

  <node name="multi_robot_launcher" pkg="bwi_launch" type="multi_robot_launcher">
    <rosparam command="load" param="configuration" file="$(find bwi_guidance_solver)/config/simulation_2.yaml" />
    <param name="launch_file" value="$(find bwi_guidance_solver)/launch/single_robot.launch.xml" />
  </node>

  <!-- Launch the person -->
  <group ns="person">
    <arg name="personid" default="person" />
    <include file="$(find bwi_gazebo_entities)/launch/controllable_person.launch">
      <arg name="x" value="5.5" />
      <arg name="y" value="23" />
      <arg name="personid" default="$(arg personid)" />
    </include>
    <node name="odom_to_pose" pkg="bwi_guidance_solver" type="odom_to_pose" />
  </group>
  <node name="person_visualizer" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find bwi_guidance_solver)/perspectives/person.perspective">
    <remap from="/image" to="/person/camera/rgb/image_raw" />
  </node>

  <!-- Launch the base robot navigator -->
  <node name="base_robot_navigator" pkg="bwi_guidance_solver" type="base_robot_navigator">
    <param name="map_file" value="$(find bwi_guidance)/maps/map3.yaml" />
    <param name="graph_file" value="$(find bwi_guidance)/maps/graph_map3.yaml" />
  </node>

</launch>
